ESLint not respecting object spread / rest syntax

## Type of Change
- [ ] New Feature Request
- [ ] Documentation
- [ ] Improvement / Suggestion
- [x] Bug
- [ ] Other (please clarify below)

## Summary
<!-- Brief summary of the issue. -->
This [syntax](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax), in the context of an evergreen application would be expected IMO so to have the linter complain seems inconsistent.

```javascript
const otherObj = {};
const myObj = {
  name: 'owen'
};

myObj = {
  ...otherObj
}
```

```shell
ERROR in ./components/header/header.js
Module Error (from ../node_modules/eslint-loader/index.js):

/Users/owenbuckley/Workspace/project-evergreen/repos/create-evergreen-app/src/components/header/header.js
  11:3  error  Parsing error: Unexpected token ..

✖ 1 problem (1 error, 0 warnings)
```

<img width="538" alt="screen shot 2018-10-31 at 6 03 17 pm" src="https://user-images.githubusercontent.com/895923/47821532-61e2e680-dd37-11e8-95b9-9cf40b516b2b.png">
<img width="911" alt="screen shot 2018-10-31 at 6 03 31 pm" src="https://user-images.githubusercontent.com/895923/47821533-61e2e680-dd37-11e8-8ac2-f7ce69c45086.png">

## Details
<!-- code snippet, steps to reproduce, etc -->
Support object rest / spread

--- a/.eslintrc
+++ b/.eslintrc
@@ -1,6 +1,6 @@
 {
   "parserOptions": {
-    "ecmaVersion": 2017,
+    "ecmaVersion": 2018,
     "sourceType": "module"
   },
   "env": {

